src/FormElement/ScheduleElement/MonthlyFields.php

```diff
@@ -0,0 +1,173 @@
+     * @param int $fields
+     *
+     * @return $this
+     */
+    public function setAvailableFields(int $fields): self
+    {
+        $this->availableFields = $fields;
+
+        return $this;
+    }
+
+    /**
+     * Set the default field/day to be selected
+     *
+     * @param int $default
+     *
+     * @return $this
+     */
+    public function setDefault(int $default): self
+    {
+        $this->default = $default;
+
+        return $this;
+    }
+
+    /**
+     * Get all the selected weekdays
+     *
+     * @return array
+     */
+    public function getSelectedDays(): array
+    {
+        $selectedDays = [];
+        foreach (range(1, $this->availableFields) as $day) {
+            if ($this->getValue("day$day", 'n') === 'y') {
+                $selectedDays[] = $day;
+            }
+        }
+
+        if (empty($selectedDays)) {
+            $selectedDays[] = $this->default;
+        }
+
+        return $selectedDays;
+    }
+
+    protected function assemble()
+    {
+        $this->getAttributes()->set('id', $this->protectId('monthly-fields'));
+
+        $runsOn = $this->getPopulatedValue('runsOn', static::RUNS_EACH);
+        $this->addElement('radio', 'runsOn', [
+            'required' => true,
+            'class'    => 'autosubmit',
+            'value'    => $runsOn,
+            'options'  => [static::RUNS_EACH => $this->translate('Each')],
+        ]);
+
+        $listItems = HtmlElement::create('ul', ['class' => ['schedule-element-fields', 'multiple-fields']]);
+        if ($runsOn === static::RUNS_ONTHE) {
+            $listItems->getAttributes()->add('class', 'disabled');
+        }
+
+        $foundCheckedDay = false;
+        foreach (range(1, $this->availableFields) as $day) {
+            $checkbox = $this->createElement('checkbox', "day$day", [
+                'class' => ['autosubmit', 'sr-only'],
+                'value' => $this->getPopulatedValue("day$day", 'n')
+            ]);
+            $this->registerElement($checkbox);
+
+            $foundCheckedDay = $foundCheckedDay || $checkbox->isChecked();
+            $htmlId = $this->protectId("day$day");
+            $checkbox->getAttributes()->set('id', $htmlId);
+
+            $listItem = HtmlElement::create('li');
+            $listItem->addHtml($checkbox, HtmlElement::create('label', ['for' => $htmlId], $day));
+            $listItems->addHtml($listItem);
+        }
+
+        if (! $foundCheckedDay) {
+            $this->getElement("day{$this->default}")->setChecked(true);
+        }
+
+        $monthlyWrapper = HtmlElement::create('div', ['class' => 'monthly']);
+        $runsEach = $this->getElement('runsOn');
+        $runsEach->prependWrapper($monthlyWrapper);
+        $monthlyWrapper->addHtml($runsEach, $listItems);
+
+        $this->addElement('radio', 'runsOn', [
+            'required' => $runsOn !== static::RUNS_EACH,
+            'class'    => 'autosubmit',
+            'options'  => [static::RUNS_ONTHE => $this->translate('On the')]
```

Comment:
This also works. Saves you the get/clear/add shenanigans below. Yes, break chain on failure can't be defined, but it's a single validator anyway ;)

```suggestion
            'options'  => [static::RUNS_ONTHE => $this->translate('On the')],
            'validators' => [new InArrayValidator([
                'strict'   => true,
                'haystack' => [static::RUNS_EACH, static::RUNS_ONTHE]
            ])]
```